src/conn/conn_api.c

```diff
@@ -2053,6 +2053,30 @@ __debug_mode_log_retention_config(WT_SESSION_IMPL *session, const char *cfg[])
     return (ret);
 }
 
+/*
+ * __debug_mode_background_compact_config --
+ *     Set the debug configurations for the background compact server.
+ */
+static int
+__debug_mode_background_compact_config(WT_SESSION_IMPL *session, const char *cfg[])
+{
+    WT_CONFIG_ITEM cval;
+    WT_CONNECTION_IMPL *conn;
+
+    conn = S2C(session);
+
+    WT_RET(__wt_config_gets(session, cfg, "debug_mode.background_compact", &cval));
+    if (cval.val) {
+        conn->background_compact.max_file_idle_time = 30;
+        conn->background_compact.max_file_skip_time = 20;
+    } else {
+        conn->background_compact.max_file_idle_time = WT_DAY;
+        conn->background_compact.max_file_skip_time = 60 * WT_MINUTE;
```

Comment:
Could you make these fixed numbers defines? Something like:

#define WT_BACKGROUND_COMPACT_MAX_IDLE_TIME_DEBUG 30
#define WT_BACKGROUND_COMPACT_MAX_SKIP_TIME_DEBUG 20
#define WT_BACKGROUND_COMPACT_MAX_IDLE_TIME WT_DAY
#define WT_BACKGROUND_COMPACT_MAX_SKIP_TIME 60 * WT_MINUTE

Also: Would you consider setting the debug version down even lower? To 1, or maybe 2? That should allow us to get test coverage faster.

Reply:
If they're too low we won't be able to test the other heuristics about whether to compact or not. We need to leave some time for compact to do work. Also the background compact server sleeps for 10 seconds after each full pass of the metadata file so we'd need to change that to make it lowed.
@etienneptl

Reply:
I've added a debug mode to the wait time after a full iteration and decreased the other times so we now have
#define WT_BACKGROUND_COMPACT_MAX_IDLE_TIME_DEBUG 10
#define WT_BACKGROUND_COMPACT_MAX_SKIP_TIME_DEBUG 5
#define WT_BACKGROUND_COMPACT_WAIT_TIME_DEBUG 2